examples/policy/opa-gatekeeper/README.md

```diff
@@ -0,0 +1,234 @@
+# User Guide: Protecting Agentic Sandboxes with OPA Gatekeeper
+
+## 1. Overview
+
+This guide provides step-by-step instructions for configuring a OPA Gatekeeper security policy on a Kubernetes cluster. The goal of this policy is to **prevent any user or process from granting new permissions to a ServiceAccount that is actively being used by a custom `Sandbox` resource.**
+
+This acts as a critical security boundary, preventing accidental or malicious privilege escalation for sandboxed environments.
+
+**How it Works:**  
+The policy intercepts all `RoleBinding` and `ClusterRoleBinding` creation requests. If the request targets a `ServiceAccount` that is referenced by a running `Sandbox`, OPA Gatekeeper will **block the request and return an error**.
+
+---
+
+## 2. Prerequisites
+
+Before you begin, ensure you have the following:
+
+- `kubectl` access to a Kubernetes **Standard** cluster with permissions to install Helm charts.
+- Helm v3+ installed on your local machine.
+- The **Sandbox controller** must already be installed. This is crucial as it provides the `Sandbox` Custom Resource Definition (CRD) that the policy needs to recognize.
+
+---
+
+## 3. Configuration Steps
+
+### Step 1: Install [OPA Gatekeeper](https://open-policy-agent.github.io/gatekeeper/website/docs/install/)
+
+Use the following Helm commands to deploy it to your cluster.
+
+```bash
+# 1. Add the official OPA Gatekeeper Helm repository
+helm repo add gatekeeper https://open-policy-agent.github.io/gatekeeper/charts
+
+# 2. Update your local Helm repositories
+helm repo update
+
+# 3. Install OPA into its own namespace
+helm install gatekeeper/gatekeeper --name-template=gatekeeper --namespace gatekeeper-system --create-namespace
+```
+
+You can verify the installation by checking the pods and CRDs in the gatekeeper-system namespace:
+
+```bash
+kubectl get pods -n gatekeeper-system
+```
+
+All pods should be in the `Running` state.
+
+### Step 2: Verify Custom Resource Definitions (CRDs)
+
+Ensure that the Gatekeeper CRDs are present in your cluster:
+
+```bash
+kubectl get crds | grep gatekeeper
+```
+
+If either CRD is not found, stop and resolve that issue before proceeding.
+
+### Step 3: Define the ConstraintTemplate
+
+The ConstraintTemplate provides the Rego logic that Gatekeeper will use to enforce the policy. This template looks for `Pods` using the targeted `ServiceAccount` and checks if they are owned by a `Sandbox` CR.
+
+Apply the template:
+
+```bash
+kubectl apply -f template-sandbox-binding.yaml
+```
+
+This creates a new CRD called `K8sPreventSandboxServiceAccountBinding` that Gatekeeper will recognize. You can verify it by running the following:
+
+```bash
+kubectl get crds | grep gatekeeper 
+
+# You should see a crd like the following:
+k8spreventactiveserviceaccountbinding.constraints.gatekeeper.sh   
```

Comment:
Why not just get that crd instead of getting all and grep?

Reply:
done.